pkgs/development/libraries/harfbuzz/default.nix

```diff
@@ -77,22 +81,23 @@ stdenv.mkDerivation (finalAttrs: {
     libintl
     pkg-config
     python3
-    glib
     gtk-doc
     docbook-xsl-nons
     docbook_xml_dtd_43
-  ] ++ lib.optional withIntrospection gobject-introspection;
+  ] ++ lib.optional withIntrospection gobject-introspection
+  ++ lib.optional withGlib glib;
 
-  buildInputs = [ glib freetype ]
+  buildInputs = [ freetype ]
+    ++ lib.optionals withGlib [ glib ]
     ++ lib.optionals withCoreText [ ApplicationServices CoreText ];
 
   propagatedBuildInputs = lib.optional withGraphite2 graphite2
-    ++ lib.optionals withIcu [ icu harfbuzz ];
```

Comment:
This was also causing a circular dependency when trying to build mingw. Perhaps I should do...
```
propagateHarfbuzz ? if stdenv.hostplatform.isMinGW then false else true
...
++ lib.optionals withIcu [ icu ]
++ lib.optionals propagateHarfbuzz [ harfbuzz ]
```

Reply:
Thoughts?

Reply:
It's clear from `withIcu ? stdenv.hostPlatform.isMinGW` that ICU support is enforced (why?) for MinGW, so just pass `(withIcu && !stdenv.hostPlatform.isMinGW)` here.

Reply:
I had trouble getting this to compile without icu.

Reply:
Added recommended chagnes